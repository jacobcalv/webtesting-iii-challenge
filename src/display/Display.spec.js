// Test away!
import React from 'react';
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import Display from './Display'

//Display Snapshot and ensures that file displays

test('should display and match snapshot', () => {
    const wrapper = rtl.render(<Display/>)
    expect(wrapper.asFragment()).toMatchSnapshot()
    wrapper.debug()
})

test('should display unlocked at start', () => {
    const wrapper = rtl.render(<Display/>)
    const unlocked = wrapper.queryByText(/unlocked/i)
    expect(unlocked).toBeVisible();
})

// test('Door Closes', async() => {
//     const wrapper = rtl.render(<Dashboard/>)
//     await wrapper.findByText(/open/i)
    
//     const close = wrapper.getByText(/close/i)
    
//     rtl.act(() => {
//         rtl.fireEvent.click(close)
//     })
//     expect(wrapper.getByText(/close/i)).not.toBe(/open/i)
// })

test('')