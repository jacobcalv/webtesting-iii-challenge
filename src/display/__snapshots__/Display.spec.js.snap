// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</DocumentFragment>
`;
